### ISSUE TITLE
MapboxCoreNavigationTests failures related with RouteController and Navigator need to fix

### ISSUE THREAD

**Issue by ShanMa1991** [mapbox/mapbox-navigation-ios#2913]
The MapboxCoreNavigationTests failed in the following tests as in [the workflow](https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-ios/2950/workflows/920fa868-d1c5-4e45-b33a-a280266b252c/jobs/29227). They seem to be related with RouteController and Navigator, which need to investigate and fix.
```
MapboxCoreNavigationTests.testDepart()
		MapboxCoreNavigationTests.testDepart()
		MapboxCoreNavigationTests.testNewStep()
		MapboxCoreNavigationTests.testNewStep()
		MapboxCoreNavigationTests.testOrderOfExecution()
		MapboxCoreNavigationTests.testOrderOfExecution()
		MapboxCoreNavigationTests.testOrderOfExecution()
		MapboxCoreNavigationTests.testShouldReroute()
		MapboxCoreNavigationTests.testShouldReroute()
		NavigationServiceTests.testGeneratingAnArrivalEvent()
		NavigationServiceTests.testLocationCourseShouldNotChange()
		NavigationServiceTests.testMultiLegRoute()
		NavigationServiceTests.testMultiLegRoute()
		NavigationServiceTests.testProactiveRerouting()
		NavigationServiceTests.testReroutingFromALocationSendsEvents()
		NavigationServiceTests.testSnappedAtEndOfStepLocationWhenCourseIsSimilar()
		NavigationServiceTests.testSnappedAtEndOfStepLocationWhenCourseIsSimilar()
		NavigationServiceTests.testSnappedAtEndOfStepLocationWhenCourseIsSimilar()
		NavigationServiceTests.testSnappedAtEndOfStepLocationWhenCourseIsSimilar()
		NavigationServiceTests.testSnappedAtEndOfStepLocationWhenCourseIsSimilar()
		NavigationServiceTests.testSnappedAtEndOfStepLocationWhenMovingSlowly()
		NavigationServiceTests.testSnappedAtEndOfStepLocationWhenMovingSlowly()
		NavigationServiceTests.testSnappedAtEndOfStepLocationWhenMovingSlowly()
		NavigationServiceTests.testSnappedAtEndOfStepLocationWhenMovingSlowly()
		NavigationServiceTests.testSnappedAtEndOfStepLocationWhenMovingSlowly()
		NavigationServiceTests.testSnappedAtEndOfStepLocationWhenMovingSlowly()
		NavigationServiceTests.testSnappedAtEndOfStepLocationWhenMovingSlowly()
		NavigationServiceTests.testSnappedLocationForUnqualifiedLocation()
		NavigationServiceTests.testSnappedLocationForUnqualifiedLocation()
		NavigationServiceTests.testSnappedLocationForUnqualifiedLocation()
		NavigationServiceTests.testSnappedLocationForUnqualifiedLocation()
		NavigationServiceTests.testSnappedLocationForUnqualifiedLocation()
		NavigationServiceTests.testSnappedLocationForUnqualifiedLocation()
		NavigationServiceTests.testUnimplementedLogging()
		NavigationServiceTests.testUserIsOffRoute()
		NavigationServiceTests.testUserIsOffRoute()
		NavigationServiceTests.testUserIsOnRoute()
		PassiveLocationDataSourceTests.testManualLocations()
```

**Comment by ShanMa1991** [comment]
These test failures are mostly related with the `func locationManager(_ manager:, didUpdateLocations:)` of `MapboxNavigationService: CLLocationManagerDelegate`

**Comment by ShanMa1991** [comment]
When each test case separately, they all pass. such as only running the `PassiveLocationDataSourceTests.testManualLocations()` or `MapboxCoreNavigationTests.testDepart()` each time, they all passed. The problem should be related with the test cycle.

**Comment by ShanMa1991** [comment]
It's fixed by #2918 . Now the `MapboxCoreNavigationTests` could build and pass all now.
